### PR TITLE
Configurable moat hexes positions

### PR DESCRIPTION
**Pull request by ArseniyShestakov**
I think it's pointless to make hardcoded fix for Fortress moat so just implemented it for faction config. 

This one break saves and mods, but mods breakage can be easily avoided. Though as far as I see faction schema don't set any default values so I added it for all towns.

### CODE REVIEW

**Comment by DjWarmonger** [comment]
Wait, what is wrong with Fortress Moat and why it's important?

**Comment by ArseniyShestakov** [comment]
Fortress moat unlike in other towns take twice more hexes.

**Comment by DjWarmonger** [comment]
I see. This could be somehow combined with special Tower mines handling, as far as I know they are still not orking correctly.

**Comment by ArseniyShestakov** [comment]
I'll look into that.

**Comment by alexvins** [comment]
No need to break saves here, but anyway update format version first.

**Comment by ArseniyShestakov** [comment]
Added saves and mods compatibility that does work. Find out that movement code in battle may need improvement to work properly with two hex moat, but that isn't really related to this branch.

File: lib/CTownHandler.cpp [hunk 1]
```diff
@@ -85,6 +86,12 @@ CTown::~CTown()
 		str.dellNull();
 }
 
+std::vector<BattleHex> CTown::defaultMoatHexes()
+{
+	static const BattleHex moatHexes[] = {11, 28, 44, 61, 77, 111, 129, 146, 164, 181};
+	return std::vector<BattleHex>(moatHexes, moatHexes + ARRAY_COUNT(moatHexes));
```
Comment (alexvins) [inline lib/CTownHandler.cpp]: initializer lists allowed now. 
```cpp
static const BattleHex moatHexes[] = {11, 28, 44, 61, 77, 111, 129, 146, 164, 181};
	return std::vector<BattleHex>(moatHexes, moatHexes + ARRAY_COUNT(moatHexes));

**Pull request merged**